pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -390,6 +424,81 @@ func handleZedrouterRestarted(ctxArg interface{}, restartCounter int) {
 	}
 }
 
+// handleAppInstanceStatusCreate - Handle AIS create. Publish AppStatusSummary to ledmanager
+func handleAppInstanceStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctx := ctxArg.(*zedmanagerContext)
+	publishAppInstanceSummary(ctx)
+}
+
+// handleAppInstanceStatusModify - Handle AIS modify. Publish AppStatusSummary to ledmanager
+func handleAppInstanceStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	ctx := ctxArg.(*zedmanagerContext)
+	publishAppInstanceSummary(ctx)
+}
+
+// handleAppInstanceStatusDelete - Handle AIS delete. Publish AppStatusSummary to ledmanager
+func handleAppInstanceStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctx := ctxArg.(*zedmanagerContext)
+	publishAppInstanceSummary(ctx)
+}
+
+func publishAppInstanceSummary(ctxPtr *zedmanagerContext) {
+
+	summary := types.AppInstanceSummary{
+		TotalApps:    0,
+		TotalRunning: 0,
+		TotalHalted:  0,
+		TotalError:   0,
+	}
+	items := ctxPtr.pubAppInstanceStatus.GetAll()
+	for _, st := range items {
+		status := st.(types.AppInstanceStatus)
+
+		config := lookupAppInstanceConfig(ctxPtr, status.Key())
+		if config == nil {
```

Comment:
I'd completely forgotten about the status.EffectiveActivate. Great that you found it. But with that you don't need to look at the config at all.

Reply:
ok